serverRoot/src/main/server/org/compiere/server/WorkflowProcessor.java

```diff
@@ -374,43 +375,55 @@ private int sendEmail (MWFActivity activity, String AD_Message,
 		int counter = 0;
 		
 		//	To Activity Owner
-		if (m_client.sendEMail(activity.getAD_User_ID(), subject, message, pdf))
-			counter++;
+		counter++;
 		list.add (new Integer(activity.getAD_User_ID()));
 
 		//	To Process Owner
 		if (toProcess
-			&& process.getAD_User_ID() != activity.getAD_User_ID())
-		{
-			if (m_client.sendEMail(process.getAD_User_ID(), subject, message, pdf))
-				counter++;
+			&& process.getAD_User_ID() != activity.getAD_User_ID()) {
+			counter++;
 			list.add (new Integer(process.getAD_User_ID()));
 		}
 		
 		//	To Activity Responsible
 		MWFResponsible responsible = MWFResponsible.get(getCtx(), activity.getAD_WF_Responsible_ID());
 		counter += sendAlertToResponsible (responsible, list, process,
-			subject, message, pdf);
+			subject, message.get(), attachmentAsPDF.get(), po);
 		
 		//	To Process Responsible
 		if (toProcess
-			&& process.getAD_WF_Responsible_ID() != activity.getAD_WF_Responsible_ID())
-		{
+			&& process.getAD_WF_Responsible_ID() != activity.getAD_WF_Responsible_ID()) {
 			responsible = MWFResponsible.get(getCtx(), process.getAD_WF_Responsible_ID());
 			counter += sendAlertToResponsible (responsible, list, process,
-				subject, message, pdf);
+				subject, message.get(), attachmentAsPDF.get(), po);
 		}
 		
 		//	Processor SuperVisor
 		if (toSupervisor 
 			&& m_model.getSupervisor_ID() != 0
-			&& !list.contains(new Integer(m_model.getSupervisor_ID())))
-		{
-			if (m_client.sendEMail(m_model.getSupervisor_ID(), subject, message, pdf))
-				counter++;
+			&& !list.contains(new Integer(m_model.getSupervisor_ID()))) {
+			counter++;
 			list.add (new Integer(m_model.getSupervisor_ID()));
 		}
-
+		Trx.run(transactionName -> {
+			//	Get instance for notifier
+			DefaultNotifier notifier = (DefaultNotifier) QueueLoader.getInstance().getQueueManager(DefaultNotifier.QUEUETYPE_DefaultNotifier)
+					.withContext(getCtx())
+					.withTransactionName(transactionName);

```

Comment:
For clean code, extract this as a private method (something like getDefaultNotifierInstance) and get rid of the comment.